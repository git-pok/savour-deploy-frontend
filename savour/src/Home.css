.Home-h1 {
  text-align: center;
  display: block;
  font-size: 60px;
  margin: 0px 0px;
  padding-top: 120px;
}

.Home-subtitle {
  text-align: center;
  display: block;
  font-size: 26px;
  margin: 0px auto -20px auto;
}

.Home-p {
  display: block;
  width: 700px;
  margin: 70px auto 0px auto;
  text-align: left;
  padding: 30px 30px;
  background-color: rgb(216, 216, 216);
  border-radius: 10px;
}

.Home-button {
  display: flex;
  justify-content: center;
  width: 700px;
  margin: 0px auto;
  text-align: left;
  padding-top: 70px;
}

@media only screen and (max-width: 725px) and (min-width: 400px) {
  .Home-h1 {
    padding-top: 20px;
  }
  
  .Home-p {
    width: 90%;
  }

  .Home-button {
    display: flex;
    justify-content: center;
    width: 100%;
    margin: 0px auto;
    text-align: left;
    padding-top: 70px;
  }
}