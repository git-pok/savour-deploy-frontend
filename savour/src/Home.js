import CenterText from './CenterText.js';
import Button from './Button.js';
import './Home.css';

/**
 * Home
 * Props: none
 * Renders home page
*/
const Home = () => {

  const pgh1 = `Savour is a recipe app that is built to
    give the user access to recipes and tools to support recipes.
    With savour, a user may search/like/dislike/try/save/share a recipe,
    create a custom version of a recipe, create a recipelist with
    recipes for an occasion, create a shoppinglist, and more. Savour
    has recipes for beginners and experts also.`;

  const pgh2 = `lunch, breakfast, sweet treats,
    dinner, storecupboard, desserts, cheese,
    fish and seafood, pasta, vegan, kids' baking, meat, chicken,
    savoury pastries, keto, low calorie, salads, vegan, baking,
    bread, cakes, quick bakes, biscuits,
    high protein, free from baking, breakfast, dinner, smoothies,
    fitness & lifestyle, lunch.`;

  return (
    <>
    <h1 className="Home-h1">Welcome to Savour â€“ The recipe app</h1>
    <div className="Home-p">
      <h2 className="Home-subtitle">What is Savour</h2>
      <CenterText text={pgh1} color="black" fontSize={23} />
    </div>
    <div className="Home-p">
      <h2 className="Home-subtitle">Recipe Categories</h2>
      <CenterText text={pgh2} color="black" fontSize={23} />
    </div>
    <div className="Home-button">
      <Button buttonText="RECIPES" link="/recipes" />
      <Button buttonText="CONTACT" link="/contact" />
    </div>
    </>
  );
}


export default Home;