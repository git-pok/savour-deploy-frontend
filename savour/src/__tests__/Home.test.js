import Home from '../Home.js';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';

test("renders Home without crashing", () => {
  render(
    <MemoryRouter>
      <Home />
    </MemoryRouter>
  );
});

test("renders snapshot without crashing", () => {
  const { asFragment } = render(
    <MemoryRouter>
      <Home />
    </MemoryRouter>
  );

  expect(asFragment()).toMatchSnapshot();
});

test("home page elements", () => {
  const { getByText } = render(
    <MemoryRouter intitialEntries={["/"]}>
      <Home />
    </MemoryRouter>
  );

  const welcome = getByText("Welcome to Savour â€“ The recipe app");
  expect(welcome).toBeInTheDocument();
});