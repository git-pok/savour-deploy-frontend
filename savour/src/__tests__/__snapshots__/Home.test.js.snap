// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders snapshot without crashing 1`] = `
<DocumentFragment>
  <h1
    class="Home-h1"
  >
    Welcome to Savour â€“ The recipe app
  </h1>
  <div
    class="Home-p"
  >
    <h2
      class="Home-subtitle"
    >
      What is Savour
    </h2>
    <p
      style="color: black; font-size: 23px;"
    >
      Savour is a recipe app that is built to
    give the user access to recipes and tools to support recipes.
    With savour, a user may search/like/dislike/try/save/share a recipe,
    create a custom version of a recipe, create a recipelist with
    recipes for an occasion, create a shoppinglist, and more. Savour
    has recipes for beginners and experts also.
    </p>
  </div>
  <div
    class="Home-p"
  >
    <h2
      class="Home-subtitle"
    >
      Recipe Categories
    </h2>
    <p
      style="color: black; font-size: 23px;"
    >
      lunch, breakfast, sweet treats,
    dinner, storecupboard, desserts, cheese,
    fish and seafood, pasta, vegan, kids' baking, meat, chicken,
    savoury pastries, keto, low calorie, salads, vegan, baking,
    bread, cakes, quick bakes, biscuits,
    high protein, free from baking, breakfast, dinner, smoothies,
    fitness & lifestyle, lunch.
    </p>
  </div>
  <div
    class="Home-button"
  >
    <div
      class="Button"
    >
      <a
        exact="true"
        href="/recipes"
      >
        <button
          class="Button-button"
        >
          RECIPES
        </button>
      </a>
    </div>
    <div
      class="Button"
    >
      <a
        exact="true"
        href="/contact"
      >
        <button
          class="Button-button"
        >
          CONTACT
        </button>
      </a>
    </div>
  </div>
</DocumentFragment>
`;
